synapse/handlers/federation_event.py

```diff
@@ -1220,6 +1245,26 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase) -> No
 
         logger.debug("_handle_marker_event: received %s", marker_event)
 
+        # TODO: Move this to a background queue
+        async def handle_marker_queue(marker_event: EventBase) -> None:
```

Comment:
During an [out of band discussion with @erikjohnston](https://docs.google.com/document/d/1JJDuPfcPNX75fprdTWlxlaKjWOdbdJylbpZ03hzo638/edit#bookmark=id.sm92fqyq7vpp), we discussed making the marker events have unique `state_keys` so that all of them resolve in the current snapshot of state.

 - Avoids having to fetch previous snapshots of state up through time to find all of the marker events
 - Also avoids state resolution conflicts where only one of the marker events win